pkg/logutil/log_util.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logutil
+
+import (
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// LogVerbosity is the level of verbosity of a particular log statement.
+// Usage: log.V(LogVerbosity).Info(...)
+type LogVerbosity = int
+
+// The default log verbosity, given to the most important log statements,
+// is 0: log.V(0).Info(...) == log.Info(...)
+//
+// HACK: zapr doesn't calculate log levels correctly, so we must set log
+// level integers incorrectly (according to logr) for now so they map to zapr's.
+const (
+	// VDebug == -1. log.V(VDebug).Info(...) == log.Debug(...)
+	VDebug LogVerbosity = iota - 1
+	_
+	// VWarn == 1. log.V(VWarn).Info(...) == log.Warn(...)
+	VWarn
+)
+
+// SDKLog is the parent logger for logs generated by operator code.
+var SDKLog = logf.Log.WithName("operator-sdk")
```

Comment:
Do we need to name the parent logger? We're naming it everywhere else. Wondering if there's a reason to prepend the name operator-sdk to all log statements?

Perhaps for the SDK pkg logs yes. But for the logger being used in the templates?

Reply:
`controller-runtime` sets their [package logger](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/runtime/log/log.go#L101) with a name, so I followed that convention. Using the `"operator-sdk"` log name we can differentiate `controller-runtime` logs from operator logs. I can remove the name if we don't want to have differentiated logs.

Reply:
@estroz The controller-runtime logs should already be different from all other logs with its own name.

What we might want to differentiate between is the SDK pkg logs and the user's logs within their reconcilers and pkgs.
Right now both logs will be prefixed with `operator-sdk`. Which doesn't differentiate between our pkg logs and the user's custom logs. 
So I would prefer to keep this unnamed, and instead keep the naming in whatever pkg the logs are being used in.

Reply:
In that case we can do away with `SDKLog` and call `logf.Log` to create loggers instead. Sound good?